adapters/excon/lib/opentelemetry/adapters/excon/adapter.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Excon
+      # The Adapter class contains logic to detect and install the Excon
+      # instrumentation adapter
+      class Adapter < OpenTelemetry::Instrumentation::Adapter
+        install do |config|
+          require_dependencies
+          config[:tracer_middleware] ||= Middlewares::TracerMiddleware
```

Comment:
As with `Rest::Client`, I'm not sure that a configuration option for the middware makes sense.